### DISCUSSION TITLE
Socket.io + node.js + IIS hangs issue

### DISCUSSION THREAD

**Discussion by Sarathmunusamy93** [socketio/socket.io#4315] [Q&A]
Hi Team,

I have implemented the socket.io in my application. I will explain my application first, after that I will explain my issue.
 1. My application is node-based application which hosted via IIS with reversed proxy mechanism.
 2. I have implemented Socket.io in my application where socket.io client connects to server with below code.

**Client Code:**
  ```
try {
  let path = window.location.pathname;
  path = path.substring( 0, path.lastIndexOf( "/" ) );
  window.SocketConnection = io && io.connect( "/", { path: path + "/socket.io" } );
  } catch( exception ) {
  console.error( exception );
  }
```

The above code will execute once, my application loads. 

**Server Code:**
  ```
// Sokect connection options.
  const SocketOptions = {
  pingTimeout: 60000,
  maxHttpBufferSize: 1024 * 1024 * 48,
  transports: [ "polling", "websocket" ]
  };
  
  // Initialize the Socket connection with required parameters.
  io = socketIO( server, SocketOptions );
```

**Problem:**
	Now the issue is, if more number clients (around 3K connection) connected to server then, IIS hangs and freezed. 

**My question:**
The provided information may be insufficient to answer my question, but it would be helpful if you answer my below questions.
1.	What is the best way to handle this kind of IIS hang/ freeze on high load?
2.	Do I need to use any special Socke.io client/server option to reduce this kind of load issue?

By,
Sarath Kumar.

**Comment by kimboslice99** [comment]
This is old but I do want to mention there are a handful of settings to look at, I would start with app pool settings. Are there any private or virtual memory limits placed on the app pool? remove them entirely if so.

I am fairly certain there are some settings for websocket buffer size, which could be worth adjusting as well.
